### ISSUE TITLE
Lines starting with `#` are consumed without a warning

### ISSUE THREAD

**Issue by YellowAfterlife** [YoYoGames/GameMaker-Bugs#6125]
### Description

Time to time causes me a bit of confusion when I copy-paste a bit of code from an extension to test something and forget to change the `#define` lines.

So, for example, if you have the following in a Create event, it will compile as if the `#define` line wasn't there
```gml
show_debug_message("one!")
#define draw
show_debug_message("two!")
```

[abug_accidental_preproc.zip](https://github.com/user-attachments/files/15589860/abug_accidental_preproc.zip)

### Steps To Reproduce

1. Start GameMaker
2. Open the attached sample
3. Try running it
4. Observe the lack of compiler errors/warnings

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.4.1 (Monthly)

### Which operating system(s) are you seeing the problem on?

Windows 11

### Which platform(s) are you having trouble building this project for?

Windows

### Are you running GameMaker from inside your Steam library?

No

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by rwkay** [comment]
Fixed in 2024.8 - now errors if `#<name>` is unknown

**Comment by mgeddesGM** [comment]
verified as of  IDE v2024.800.0.588  Runtime v2024.800.0.612
